main_board/src/runner/runner.c

```diff
@@ -311,6 +315,20 @@ handle_reboot_message(job_t *job)
         job_ack(orb_mcu_Ack_ErrorCode_RANGE, job);
         LOG_ERR("Reboot with delay > 60 seconds: %u", delay);
     } else {
+        // Send out shutdown scheduled CAN message
+        orb_mcu_main_ShutdownScheduled shutdown;
+        shutdown.shutdown_reason =
+            orb_mcu_main_ShutdownScheduled_ShutdownReason_JETSON_REQUESTED_REBOOT;
+        shutdown.has_ms_until_shutdown = true;
+        shutdown.ms_until_shutdown = delay * 1000;
+        publish_new(&shutdown, sizeof(shutdown),
+                    orb_mcu_main_McuToJetson_shutdown_tag,
+                    CONFIG_CAN_ADDRESS_DEFAULT_REMOTE);
+#ifdef CONFIG_MEMFAULT
+        MEMFAULT_REBOOT_MARK_RESET_IMMINENT(
+            kMfltRebootReason_JetsonRequestedReboot);
+#endif
```

Comment:
we should do all of this only if `reboot()` returns `RET_SUCCESS` below. 

Otherwise, the reboot command failed and thus the event shouldn't be sent to the Jetson nor marked to Memfault.